src/lru.c

```diff
@@ -351,6 +351,24 @@ void destroy_lru(Queue *queue)
 	destroy_queue(queue);
 }
 
+void flush_lru(Queue *queue)
+{
+	if (queue == NULL)
+		return;
+
+	dump_queue_stats(queue);
+
+	while (queue->count)
+		dequeue(queue);
+
+	// Clear statistics
+	queue->count = 0;
```

Comment:
I'm not sure that its necessary to clear the statistics. Just because the database was updated doesn't mean that past performance needs to be forgotten. The idea is to give some overall performance information so that queue sizes might be adjusted if necessary. This is normally reported as part of shutdown so that you can see how the whole session went. Which also makes me wonder if dumping the queue stats on update is necessary. It would go out on its own without being part of the larger set of statistics that gets written on shutdown.